unstract/connectors/src/unstract/connectors/filesystems/sftp/sftp.py

```diff
@@ -79,3 +98,45 @@ def test_credentials(self) -> bool:
                 "please check the connection settings."
             )
         return True
+
+    def create_dir_if_not_exists(self, input_dir: str) -> None:
+        """Method to create dir of a connector if not exists.
+
+        Args:
+            input_dir (str): input directory of source connector
+        """
+        fs_fsspec = self.get_fsspec_fs()
+        try:
+            is_dir = fs_fsspec.isdir(input_dir)
+            if not is_dir:
+                fs_fsspec.mkdir(input_dir)
+        except PermissionError as e:
+            self.raise_permission_error(input_dir=input_dir, error=e)
+
+    def upload_file_to_storage(self, source_path: str, destination_path: str) -> None:
+        """Method to upload filepath from tool to sftp connector directory.
+
+        Args:
+            source_path (str): local path of file to be uploaded, coming from tool
+            destination_path (str): target path in the storage where the file will be
+            uploaded
+        """
+        try:
+            normalized_path = os.path.normpath(destination_path)
+            fs = self.get_fsspec_fs()
+            with open(source_path, "rb") as source_file:
+                fs.write_bytes(normalized_path, source_file.read())
```

Comment:
NIT: @kirtimanmishrazipstack to avoid code duplication, use the base class method directly. I see that we are only wrapping it with some error handling.
```suggestion
            super().upload_file_to_storage(source_path, destination_path)
```
Do check and add this suggestion to the other `create_dir_if_not_exists` method as well